library_samples/Python/ocs_sample_library_preview/DataViews.py

```diff
@@ -311,11 +311,12 @@ def getDataInterpolated(
         nextPage = None
         firstPage = None
 
-        if "NextPage" in response.headers:
-            nextPage = response.headers["NextPage"]
-
-        if "FirstPage" in response.headers:
-            firstPage = response.headers["FirstPage"]
+        links = response.headers.get("Link", None)
+        if links:
```

Comment:
Looking at the format of the header, it seems like this code could be safer if it didn't assume specific indices were specific links. The header returns rel="first" and rel="next" presumably for a reason, after all. 

Could we instead do `links.split(",")`, then iterate over the results, and if the result ends in `rel="first"` we know it's the 'first' link and if the result ends in `rel="next"`, we know it's the next link. 

Then we can either split by by the `;` character and trim out the `<>`, or simply regex out the URL that's between the `<>`. 

If we can assume the links always come back in this order and there's never a "next" link with no "first" link, this is OK.

Reply:
I checked with Daphne and indeed the links can be in any order. I'll modify accordingly.

Reply:
Robust parsing in last commit

Reply:
Yeah, this looks great now.